utils/src/main/java/software/amazon/awssdk/utils/async/InputStreamSubscriber.java

```diff
@@ -125,6 +128,11 @@ public int read(byte[] bytes, int off, int len) {
     @Override
     public void close() {
         synchronized (subscribeLock) {
+            // If it is done, no-op
+            if (inputStreamState.get().equals(State.STREAMING_DONE)) {
```

Comment:
The SDK will close the input stream once a request is finished and in this case, if we already have all data buffered and received, there's no need to raise exception (exception creation is expensive)